classes/Product.php

```diff
@@ -8034,4 +8034,21 @@ public function updateDefaultSupplierData(int $idSupplier, string $supplierRefer
             )
         );
     }
+
+    /**
+     * Get Product ecotax
+     *
+     * @return ecotax
+     */
+    public function getEcotax($precision = null, $include_tax = true)
+    {
+        $precision = $precision ? $precision : Context::getContext()->currency->precision;
+        $ecotax_rate = $include_tax ? (float) Tax::getProductEcotaxRate() : 0;
+
+        return  Tools::ps_round(
+            (float) $this->ecotax * (1 + $ecotax_rate / 100),
```

Comment:
```suggestion
            (float) $this->ecotax * ((1 + $ecotax_rate) / 100),
```
You mean this?

Reply:
J'ai commité par erreur ! J'ai suis pas d'accord

Reply:
@PierreRambaud  it's correct
```php
$this->ecotax * (1 + $ecotax_rate / 100)
```